### PR TITLE
RHDEVDOCS-4299: Added a procedure to create an event sink

### PR DESCRIPTION
**Pull request by shipsing**
**DO NOT MERGE THIS PR** as the placment of the new content is under discussion.

Purpose: To resolve this issue: https://issues.redhat.com/browse/RHDEVDOCS-4299

Aligned team: DevConsole

OCP version this PR applies to: 4.12

Content for preview: [Download this assembly to see the preview in your browser](https://drive.google.com/file/d/1f7h-STSByrGnSucM47_Q428PGiznzZFE/view?usp=sharing)

SME review completed: @invincibleJai  @pierDipi 

QE review completed: @mgencur  

Peer review completed: @gabriel-rh @Preeticp

### CODE REVIEW

**Comment by shipsing** [comment]
label-do-not-merge

**Comment by shipsing** [comment]
@vikram-raj, Could you please help with SME review of this PR and share your feedback?

Thanks

**Comment by matzew** [comment]
LGTM

**Comment by invincibleJai** [comment]
UI flow looks good to me

LGTM

File: modules/serverless-creating-a-kafka-event-sink.adoc [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+[id="serverless-creating-a-Kafka-event-sink_{context}"]
+
+= Creating a Kafka event sink
+
+As a developer, you can create an event sink to receive events from a particular source, such as a bootstrap server and send them to a kafka topic. 
+
+.Prerequisites
+* You have installed the Red Hat OpenShift Serverless operator, with Knative Serving, Knative Eventing, and Knative Kafka APIs, from the Operator Hub.
+
+* Your cluster administrator has added the KafkaSink resource, provided by Red Hat, in the *Developer Catalog*.
```
Comment (pierDipi) [inline modules/serverless-creating-a-kafka-event-sink.adoc:15]: > * Your cluster administrator has added the KafkaSink resource, provided by Red Hat, in the *Developer Catalog*.

what does this mean in practice? isn't installing Knative enough to get it in the catalog?

File: modules/serverless-creating-a-kafka-event-sink.adoc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+.Procedure
+. In the developer perspective, navigate to the *+Add* view.
+. Click *Event Sink* in the *Eventing catalog*.
+. Search for KafkaSink in the catalog items and click it.
+. Click *Create Event Sink*.
+. In the form view, type the URL of the bootstrap server, which is a combination of host name and port. 
++
+image::create-event-sink.png[]
+
+. Type the name of the topic to send event data.
```
Comment (pierDipi) [inline modules/serverless-creating-a-kafka-event-sink.adoc:26]: should we add that the topic must exist as a prerequisite?

File: modules/serverless-creating-a-kafka-event-sink.adoc (hunk 1 above)
Comment (pierDipi) [reply modules/serverless-creating-a-kafka-event-sink.adoc:15]: cc @invincibleJai

Comment (shipsing) [reply modules/serverless-creating-a-kafka-event-sink.adoc:15]: @invincibleJai @vikram-raj, Should we remove this prerequisite point?

Comment (vikram-raj) [reply modules/serverless-creating-a-kafka-event-sink.adoc:15]: Yes, we can remove this. Setting up knative with Knative Kafka is enough. It will show the KafkaSink in the catalog.

**Comment by shipsing** [comment]
Hello @pierDipi @vikram-raj, Could you please verify the changes and provide an ACK so that i can proceed with other rounds of review?
Thanks

**Review by pierDipi — APPROVED** [review]
LGTM

**Comment by shipsing** [comment]
@sanketpathak Could you please help with QE review of this PR?
Thanks

File: modules/serverless-creating-a-kafka-event-sink.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// * serverless/develop/serverless-event-sinks.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-creating-a-Kafka-event-sink_{context}"]
+
+= Creating a Kafka event sink
+
+As a developer, you can create an event sink to receive events from a particular source, such as a bootstrap server and send them to a kafka topic.
```
Comment (mgencur) [inline modules/serverless-creating-a-kafka-event-sink.adoc:10]:
```suggestion
As a developer, you can create an event sink to receive events from a particular source and send them to a kafka topic. 
```

Comment (mgencur) [reply modules/serverless-creating-a-kafka-event-sink.adoc:10]: I'd leave out the "bootstrap server" out because it's misleading. The KafkaSink is there for receiving events and sending them to a Kafka cluster that is identified by bootstrap servers and the target kafka topic. It's NOT supposed to get events "from" bootstrap servers.

The rest of this PR looks good.

Comment (shipsing) [reply modules/serverless-creating-a-kafka-event-sink.adoc]: Thank you @mgencur for your suggestion. Incorporated it, please verify the same and provide an ACK.

**Comment by mgencur** [comment]
LGTM from QE

File: modules/serverless-creating-a-kafka-event-sink.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// * serverless/develop/serverless-event-sinks.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-creating-a-Kafka-event-sink_{context}"]
```
Comment (gabriel-rh) [inline modules/serverless-creating-a-kafka-event-sink.adoc:6]: Maybe use lower-case kafka in the id? - that way, it matches the file name

[id="serverless-creating-a-kafka-event-sink_{context}"]

File: modules/serverless-creating-a-kafka-event-sink.adoc [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+= Creating a Kafka event sink
+
+As a developer, you can create an event sink to receive events from a particular source and send them to a kafka topic.
```
Comment (gabriel-rh) [inline modules/serverless-creating-a-kafka-event-sink.adoc:10]: Here I think Kafka with a capital K is needed:

"send them to a Kafka topic"

File: modules/serverless-creating-a-kafka-event-sink.adoc [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+.Prerequisites
+* You have installed the Red Hat OpenShift Serverless operator, with Knative Serving, Knative Eventing, and Knative Kafka APIs, from the Operator Hub.
+
+* You have created a Kafka topic in your Kafka environment.
+
+.Procedure
+. In the developer perspective, navigate to the *+Add* view.
+. Click *Event Sink* in the *Eventing catalog*.
+. Search for KafkaSink in the catalog items and click it.
```
Comment (gabriel-rh) [inline modules/serverless-creating-a-kafka-event-sink.adoc:20]: Maybe add ticks here for `KafkaSink`? 

. Search for \`KafkaSink\` in the catalog items and click it.

**Review by gabriel-rh — COMMENTED** [review]
Looks very good! - just some very minor suggestions

File: modules/serverless-creating-a-kafka-event-sink.adoc [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+. In the form view, type the URL of the bootstrap server, which is a combination of host name and port. 
++
+image::create-event-sink.png[]
+
+. Type the name of the topic to send event data.
+. Type the name of the event sink.
+. Click *Create*. 
+
+.Verification
+. In the developer perspective, navigate to the *Topology* view.
```
Comment (Preeticp) [inline modules/serverless-creating-a-kafka-event-sink.adoc:31]: s/developer perspective/Developer perspective should be in bold. Anything that you see in the GUI should be in Bold. If a user needs to fill up a term, say in a form or something, it should be in back ticks

File: modules/serverless-creating-a-kafka-event-sink.adoc (hunk 6 above)
Comment (Preeticp) [reply modules/serverless-creating-a-kafka-event-sink.adoc:20]: Anything that you see in the GUI should be in Bold. If a user needs to fill up a term, say in a form or something, it should be in back ticks. So if KafkaSink is an option in a drop-down, maybe reword it to - Search for *KafkaSink* in the catalog items list...
If the user needs to type it to search for it then as Gabriel said, use backticks.

File: modules/serverless-creating-a-kafka-event-sink.adoc (hunk 5 above)
Comment (Preeticp) [reply modules/serverless-creating-a-kafka-event-sink.adoc:10]: +1 the mention in prereq is with a capital K

**Review by Preeticp — COMMENTED** [review]
Overall looks great @shipsing have left a couple of suggestions PTAL.

**Comment by shipsing** [comment]
Thank you @gabriel-rh and @Preeticp for your suggestions. I have incorporated all of them.
